### ISSUE TITLE
feat(research): integrate preflight, FX conversion, and preprocessing pipeline into load_data() flow

### ISSUE THREAD

**Issue by SilvioBaratto** [SilvioBaratto/optimizer#522]
## Goal
Wire all Cycle 1 components together inside `research/stock_selection_pipeline.py` so a single end-to-end run produces a clean linear-return panel in EUR base currency on the investable subset, with zero `NaN` and the §0/§1/§2/§3 acceptance gates enforced. This is the integration issue that consumes the four upstream tasks.

## Context
Confirmed by codebase trace:

- `load_data()` at `research/stock_selection_pipeline.py:192-209` returns `(DataAssembly, country_map)` today. After this issue it must additionally apply FX conversion to `assembly.prices` so downstream code sees EUR-denominated prices.
- `screen_investable()` at `research/stock_selection_pipeline.py:217-229` already returns `pd.Index` of investable tickers; after #520 lands, it raises on `< 200` and warns outside `[300, 1500]`.
- `main()` at `research/stock_selection_pipeline.py` (further down the file — see `Step 1` / `Step 2` panels) is the orchestrator that calls `load_data` then `screen_investable`. After Cycle 1, `main()` also calls `prices_to_returns(...)` and `pipeline.fit_transform(returns)` to materialise the preprocessed return panel. **For Cycle 1 scope, the preprocessed panel is logged and stashed for Cycle 2 — it is NOT yet consumed by factor construction or optimization.**
- `DataAssembly` is mutable (regular class). Direct attribute assignment `assembly.prices = fx_prices` is permitted.

## Agent and skills
- **Agents:** `python-pro`
- **Active skills:** `solid-principles`, `python-expert`

## Acceptance criteria
- [ ] `load_data()` calls `_run_db_preflight(db_manager)` (from #519) immediately after `db_manager.initialize()`.
- [ ] `load_data()` calls `apply_fx_to_prices(...)` (from #521) immediately after `assemble_all(...)` and assigns the result back to `assembly.prices`. The returned `assembly` carries EUR-denominated prices.
- [ ] `load_data()` returns `(assembly, country_map)` with the same tuple shape as today.
- [ ] `main()` after `screen_investable()` invokes `prices_to_returns(assembly.prices.loc[:, list(investable)])`, then `build_return_preprocessing_pipeline(assembly.sector_mapping).fit_transform(returns)`. The resulting DataFrame has zero `NaN` and zero `inf` and is logged (panel size, date range, ticker count) via the rich `console`.
- [ ] `assembly.assembly_hash` (from #518) is non-empty after `load_data` returns and is logged.
- [ ] When `_run_db_preflight` raises, the run aborts BEFORE any optimization output is written. Verified by an integration-style smoke test (mock the DB session to fail Check 1 and assert `RuntimeError` propagates out of `load_data`).
- [ ] `DataAssembly.n_tickers >= 2_000` and `DataAssembly.n_trading_days >= 1_260` are asserted at the end of `load_data` (one consolidated runtime check; raise `RuntimeError` if either fails).
- [ ] `main()` uses the module-level `REGION_MAP` import (from #518) where it currently relies on the inline `_validate_checklist` dict — leave the duplicate dict in `_validate_checklist` untouched (Cycle 4 owns checklist refactor); only replace usages outside `_validate_checklist` if any exist.
- [ ] All tests pass.
- [ ] SOLID, clean code (methods < 10 lines, classes < 50 lines, files < 500-600 lines), TDD.

## Dependencies
- #518: feat(data-assembly): add REGION_MAP and assembly_hash to cli/data_assembly.py
- #519: feat(research): DB pre-flight health checks in load_data() (§0)
- #520: feat(research): enforce universe screening floor and band in screen_investable() (§2)
- #521: feat(research): FX pre-conversion + sklearn return preprocessing pipeline (§1, §3)

## Implementation notes
- **Modify:** `research/stock_selection_pipeline.py` `load_data()` (~lines 192-209) — sequence: `db_manager.initialize()` → `_run_db_preflight(db_manager)` → `assemble_all(db_manager, include_delisted=True)` → `apply_fx_to_prices(assembly.prices, assembly.currency_map, assembly.fx_rates, base_currency="EUR")` → assign back to `assembly.prices` → `_assert_assembly_size(assembly)` (private helper, raises if `n_tickers < 2000` or `n_trading_days < 1260`) → `_build_country_map(db_manager)` → return.
- **Modify:** `research/stock_selection_pipeline.py` `main()` after `screen_investable(assembly)` returns the `pd.Index` of investable tickers — slice prices, run `prices_to_returns()`, build pipeline, `fit_transform`, log. Stash the cleaned return panel in a local variable named `clean_returns` for Cycle 2 to consume.
- Log the `assembly.assembly_hash` once via `console.print(...)` so the rich panel output makes it visible.
- Keep `load_data` under 30 lines — extract `_assert_assembly_size` as a private helper if needed.
- **Do NOT** add CLI flags here. `--base-currency`, `--robust`, `--persist`, `--start-date`, `--end-date`, `--seed` are Cycle 5 scope.
- **Do NOT** modify the existing factor construction / optimization code paths. Cycle 1 stops at producing `clean_returns`; Cycle 2 picks it up.
- Smoke test: add `tests/research/test_load_data_integration.py` (create dir if needed) that monkey-patches `_run_db_preflight` and `assemble_all` to return synthetic data, then asserts `load_data()` returns `(DataAssembly, dict)` with FX-converted prices and a non-empty `assembly_hash`. Use SQLite in-memory or pure mocks — do not require a live PostgreSQL.
- Verify the `make all` gate (lint + typecheck + test) is green.

**Comment by SilvioBaratto** [comment]
REVIEW: Issue body is empty. As the integration issue, the body must call out three things explicitly:

1. **Dependency chain.** Block #522 on #519 (preflight), #520 (screening floor/band), #521 (FX pre-conversion + preprocessing), and ideally #518 (assembly_hash for run-identity logging). The current title gives no ordering signal — the agent picking up the cycle could attempt #522 first and produce a half-wired diff.

2. **🔴 Double-FX-conversion guard.** `optimizer/pipeline/_orchestrator.py:377-409` runs FX conversion when `fx_config.mode != NONE` AND `currency_map`/`fx_rates` are non-None. After #521 pre-converts in `load_data()`, the call in `optimize_portfolio()` (`research/stock_selection_pipeline.py:494-514`) must drop `currency_map=assembly.currency_map` and `fx_rates=assembly.fx_rates` (or set `fx_config.mode=NONE`). Add an integration test asserting the pre-converted single-pass run matches the legacy two-stage run on weights and net Sharpe. Without this guard, returns get FX-multiplied twice and the bug surfaces only in the magnitude of returns, which downstream Sharpe normalization can hide.

3. **Ordering inside `load_data()`.** Spell out the exact pipeline:
   ```
   _preflight(db_manager)                      # #519
   assembly = assemble_all(...)                 # existing
   assembly.prices = fx_pipeline.fit_transform(assembly.prices)   # #521 §1
   returns = preprocess_pipeline.fit_transform(prices_to_returns(assembly.prices))  # #521 §3
   ```
   And clarify whether `screen_investable()` (gated by #520 floor) runs against pre- or post-FX prices (it should run against base-currency prices for consistent ADDV thresholds).

4. **TDD anchor.** A single end-to-end test that runs `main()` against a small fixture DB and asserts: preflight passed, FX conversion ran exactly once, preprocessing fitted on returns, screening floor enforced, weights sum to 1.0.

**Comment by SilvioBaratto** [comment]
Implemented.

**Modified:**
- `research/stock_selection_pipeline.py`
  - Imports added: `apply_fx_to_prices`, `build_return_preprocessing_pipeline` from `research._preprocessing`.
  - New constants `_MIN_ASSEMBLY_TICKERS = 2_000`, `_MIN_ASSEMBLY_TRADING_DAYS = 1_260`.
  - New private helper `_assert_assembly_size(assembly)` — raises `RuntimeError` naming the failing field when either floor is breached.
  - New private helper `_materialise_clean_returns(assembly, investable)` — slices EUR prices to investable columns, calls `prices_to_returns(...)`, builds the four-step return preprocessing pipeline via `build_return_preprocessing_pipeline(assembly.sector_mapping)`, runs `fit_transform`, asserts zero NaN/inf, logs `(days, tickers, date range)` to the rich console.
  - `load_data()` flow is now: `db_manager.initialize() → _run_db_preflight(db_manager) → assemble_all(...) → assembly.prices = apply_fx_to_prices(assembly.prices, assembly.currency_map, assembly.fx_rates, base_currency="EUR") → _assert_assembly_size(assembly) → _build_country_map(db_manager) → return`. Console panel logs ticker count, trading days, EUR base, and `assembly.assembly_hash`.
  - `main()` now calls `_materialise_clean_returns(assembly, investable)` immediately after `screen_investable(...)` and stashes the result in `clean_returns` for Cycle 2 to consume. Existing factor / optimization / report code paths are untouched.

**Added tests:**
- `tests/research/test_load_data_integration.py` — 10 tests covering: `load_data()` returns `(DataAssembly, dict)` with `"AAA"` country mapping, FX conversion applied (USD ticker × 0.9 rate), non-empty `assembly_hash` propagation, `RuntimeError` propagates when `_run_db_preflight` raises (verified `assemble_all` not called), `RuntimeError` raised when `n_tickers < 2_000` or `n_trading_days < 1_260`, plus direct unit tests on `_assert_assembly_size` (below/at floors) and `_materialise_clean_returns` (zero NaN/inf, columns match investable subset).

**Tests:** 10/10 new pass. Full suite: 2015 passed, 1 xpassed (was 2005 pre-#522 — additive). `mypy optimizer/` clean. `ruff check` on touched files clean.

Cycle 1 stops at `clean_returns`. Cycle 2 owns factor / optimization integration.

Not closing — Phase 5 owns acceptance verification.

**Comment by SilvioBaratto** [comment]
Verified in final review: load_data() wires _run_db_preflight → assemble_all → apply_fx_to_prices(EUR) → _assert_assembly_size (n_tickers≥2000, n_trading_days≥1260) → _build_country_map. main() materialises clean_returns via prices_to_returns + pipeline.fit_transform with NaN/inf assertion. assembly_hash logged via rich console. 7 integration tests pass.
